### PR TITLE
refactor(datasource/aws): Strict null checks

### PR DESCRIPTION
**Pull request by zharinov**
## Changes:

- Fix null check errors
- Refactor mocking

## Context:

- Blocked by #13416
- Blocked by #13454
- Blocked by #13455

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/datasource/aws-machine-image/index.spec.ts [hunk 1]
```diff
@@ -131,11 +131,15 @@ const mock1Image: DescribeImagesResult = {
 
 const mockEmpty: DescribeImagesResult = {};
 
+function mockDescribeImagesCommand(result: DescribeImagesResult): void {
+  ec2Mock.reset();
+  ec2Mock.on(DescribeImagesCommand as never).resolves(result as never);
```
Comment (zharinov) [inline lib/datasource/aws-machine-image/index.spec.ts:136]: Type error too hard for me to decode 🤷‍♂️

Comment (rarkins) [reply lib/datasource/aws-machine-image/index.spec.ts:136]: @viceice are you ok with this?

File: lib/datasource/aws-machine-image/index.ts [hunk 2]
```diff
@@ -77,13 +78,13 @@ export class AwsMachineImageDataSource extends Datasource {
         (image) => image.ImageId === newValue
       );
       if (newValueMatchingImages.length === 1) {
-        return newValueMatchingImages[0].Name;
+        return newValueMatchingImages[0].Name ?? null;
```
Comment (viceice) [inline lib/datasource/aws-machine-image/index.ts:81]: Strange, `Name` can be null?

**Comment by viceice** [comment]
Needs test fixes too

File: lib/datasource/aws-machine-image/index.ts (hunk 2 above)
Comment (zharinov) [reply lib/datasource/aws-machine-image/index.ts:81]: Yes, according to type definition

File: lib/datasource/aws-machine-image/index.ts [hunk 3]
```diff
@@ -52,16 +52,17 @@ export class AwsMachineImageDataSource extends Datasource {
...
+      const ts1 = image1.CreationDate ? Date.parse(image1.CreationDate) : 0;
+      const ts2 = image2.CreationDate ? Date.parse(image2.CreationDate) : 0;
+      return ts1 - ts2;
+    });
   }
 
   @cache({
     namespace: `datasource-${AwsMachineImageDataSource.id}`,
-    key: ({ registryUrl, lookupName }: GetReleasesConfig, newValue: string) =>
-      `getDigest:${registryUrl}:${lookupName}:${newValue ?? ''}`,
```
Comment (zharinov) [inline lib/datasource/aws-machine-image/index.ts:64]: `registryUrl` always is `undefined` for this datasource, isn't it?

Comment (rarkins) [reply lib/datasource/aws-machine-image/index.ts:64]: Yes I think so

Comment (zharinov) [reply lib/datasource/aws-machine-image/index.ts:64]: 👍

File: lib/datasource/aws-machine-image/index.spec.ts (hunk 1 above)
Comment (viceice) [reply lib/datasource/aws-machine-image/index.spec.ts:136]:
```suggestion
  ec2Mock.on(DescribeImagesCommand).resolves(result);
```

Works for me, no warnings or anything else. 🤔

**Review by rarkins — APPROVED** [review]

**Pull request merged**